redis/credentials.py

```diff
@@ -0,0 +1,50 @@
+from typing import Callable, Optional
+
+
+class CredentialsProvider:
+    def __init__(
+        self,
+        username: str = "",
```

Comment:
In theory the default can be None, so you may want a Union here. Both for username and password